t/cli.c

```diff
@@ -540,10 +546,35 @@ int main(int argc, char **argv)
     }
     argc -= optind;
     argv += optind;
-    if ((ctx.certificates.count == 0) != (ctx.sign_certificate == NULL)) {
+
+    if (raw_pub_key_file != NULL) {
+        int is_dash = !strcmp(raw_pub_key_file, "-");
+        if (is_server) {
+            ctx.certificates.list = malloc(sizeof(*ctx.certificates.list));
+            load_raw_public_key(ctx.certificates.list, is_dash ? cert_location : raw_pub_key_file);
+            ctx.certificates.count = 1;
+        } else if (!is_dash) {
+            ptls_iovec_t raw_pub_key;
+            EVP_PKEY *pubkey;
+            load_raw_public_key(&raw_pub_key, raw_pub_key_file);
+            pubkey = d2i_PUBKEY(NULL, (const unsigned char **)&raw_pub_key.base, raw_pub_key.len);
+            if (pubkey == NULL) {
+                fprintf(stderr, "Failed to create an EVP_PKEY from the key found in %s\n", raw_pub_key_file);
+                return 1;
+            }
+            setup_raw_pubkey_verify_certificate(&ctx, pubkey);
+        }
+        ctx.use_raw_public_keys = 1;
+    } else {
+        if (cert_location)
+            load_certificate_chain(&ctx, cert_location);
+    }
+
+    if (!ctx.use_raw_public_keys && (ctx.certificates.count == 0) != (ctx.sign_certificate == NULL)) {
```

Comment:
I think we can revert the addition of `!ctx.use_raw_public_keys &&` now that the server stores the raw public key in `ctx.certificates`?
```suggestion
    if ((ctx.certificates.count == 0) != (ctx.sign_certificate == NULL)) {
```